### PR TITLE
Release/v1.3.1

### PR DESCRIPTION
**Pull request by djkwagala**
Includes fix to:
1. Disable click execution while an action is running #15
2. Disable loading of new data while resizing in web-modeler

### CODE REVIEW

File: src/BarChart/components/BarChartDataHandler.ts [hunk 1]
```diff
@@ -113,12 +117,28 @@ export class BarChartDataHandler extends Component<BarChartDataHandlerProps> {
...
-        if (options.mxObject) {
-            handleOnClick(options.options, options.mxObject, options.mxForm);
-        } else if (options.trace) {
-            this.createDataPoint(options.options, options.trace)
-                .then(mxObject => handleOnClick(options.options, mxObject, options.mxForm))
-                .catch(error => mx.ui.error(`An error occured while creating ${options.options.dataEntity} object: ${error}`));
+        if (!this.isRunningAction) {
+            this.onStartAction();
+            if (options.mxObject) {
+                // handleOnClick(options.options, options.mxObject, options.mxForm);
```
Comment (Andries-Smit) [inline src/BarChart/components/BarChartDataHandler.ts]: old comment

File: src/HeatMap/components/HeatMapDataHandler.ts [hunk 2]
```diff
@@ -168,6 +186,22 @@ class HeatMapDataHandler extends Component<HeatMapDataHandlerProps> {
...
+            }
+        } , 120);
+    }
+    private onStopAction() {
+        this.isRunningAction = false;
+        if (this.showProgress) {
+            mx.ui.hideProgress(this.showProgress);
+        }
+
+    }
```
Comment (Andries-Smit) [inline src/HeatMap/components/HeatMapDataHandler.ts:204]: Remove empty line

File: src/HeatMap/components/HeatMapDataHandler.ts [hunk 3]
```diff
@@ -121,16 +124,31 @@ class HeatMapDataHandler extends Component<HeatMapDataHandlerProps> {
...
+                        mx.ui.error(error);
+                        this.onStopActionbound();
+                    });
             } else {
                 this.createDataPoint(options.options, options.trace)
-                    .then(newMxObject => handleOnClick(options.options, newMxObject, options.mxForm))
+                    .then(newMxObject => {
+                        handleOnClick(options.options, newMxObject, options.mxForm)
+                            .then(this.onStopActionbound)
+                            .catch((error) => {
```
Comment (Andries-Smit) [inline src/HeatMap/components/HeatMapDataHandler.ts]: no need for parentheses around (error)

File: src/HeatMap/components/HeatMapDataHandler.ts [hunk 4]
```diff
@@ -168,6 +186,22 @@ class HeatMapDataHandler extends Component<HeatMapDataHandlerProps> {
...
+
+    private onStartAction() {
+        this.isRunningAction = true;
+        setTimeout(() => {
+            if (this.isRunningAction) {
+                this.showProgress = mx.ui.showProgress();
+            }
+        } , 120);
+    }
+    private onStopAction() {
```
Comment (Andries-Smit) [inline src/HeatMap/components/HeatMapDataHandler.ts]: empty line before function declaration

File: src/ColumnChart/ColumnChart.webmodeler.ts [hunk 5]
```diff
@@ -15,8 +15,9 @@ import BarChartContainerProps = Container.BarChartContainerProps;
 
 // tslint:disable-next-line class-name
 export class preview extends Component<BarChartContainerProps, { updatingData: boolean }> {
-    state = { updatingData: true };
+    readonly state = { updatingData: true };
     private instanceID = getInstanceID(this.props.friendlyId, store, "bar");
+    private scatterData = this.getData(this.props);
```
Comment (Andries-Smit) [inline src/ColumnChart/ColumnChart.webmodeler.ts:20]: When new series are added in the WM will they all have new data?

File: src/utils/data.ts [hunk 6]
```diff
@@ -273,39 +274,46 @@ export const fetchByREST = (url: string): Promise<any> => new Promise((resolve,
     }).catch(error => reject(`${errorMessage} ${error.message}`));
 });
 
-export const handleOnClick = <T extends EventProps>(options: T, mxObject?: MxO, mxform?: mxui.lib.form._FormBase) => {
+export const handleOnClick = <T extends EventProps>(options: T, mxObject?: MxO, mxform?: mxui.lib.form._FormBase): Promise<Callback> => new Promise((resolve, reject) => {
     const context = new mendix.lib.MxContext();
 
     if (!mxObject || options.onClickEvent === "doNothing") {
-        return;
+        resolve();
```
Comment (Andries-Smit) [inline src/utils/data.ts]: should we resolve and return?

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/utils/data.ts (hunk 6 above)
Comment (djkwagala) [reply src/utils/data.ts]: Yea, right! thanks

File: src/HeatMap/components/HeatMapDataHandler.ts (hunk 4 above)
Comment (djkwagala) [reply src/HeatMap/components/HeatMapDataHandler.ts]: Thanks for catching it. (uuh, I have references of usage on top of every function, they mislead, however my linters should do their job - will fix them)

File: src/HeatMap/components/HeatMapDataHandler.ts (hunk 3 above)
Comment (djkwagala) [reply src/HeatMap/components/HeatMapDataHandler.ts]: sure, thanks

File: src/HeatMap/components/HeatMapDataHandler.ts (hunk 2 above)
Comment (djkwagala) [reply src/HeatMap/components/HeatMapDataHandler.ts:204]: removed

File: src/BarChart/components/BarChartDataHandler.ts (hunk 1 above)
Comment (djkwagala) [reply src/BarChart/components/BarChartDataHandler.ts]: removed

**Pull request closed without merging**